app-modules/service-management/src/Filament/Resources/ServiceRequestResource/Pages/ListServiceRequests.php

```diff
@@ -166,11 +167,38 @@ public function table(Table $table): Table
                         )
                     ))
                     ->preload(),
-                Filter::make('Unassigned')
+                SelectFilter::make('assignedTo')
+                    ->searchable()
+                    ->preload()
+                    ->options(fn () => [
+                        '' => ['unassigned' => 'Unassigned'],
+                        'Assigned to' => User::query()->limit(50)->pluck('name', 'id')->toArray(), // Initially load 50 users
+                    ])
+                    ->getSearchResultsUsing(fn (string $search): array => ['Assigned To' => User::query()->where(new Expression('lower(name)'), 'like', '%' . strtolower($search) . '%')->take(50)->pluck('name', 'id')->toArray()])
                     ->query(
-                        fn (Builder $query) => $query->whereDoesntHave('assignedTo', function (Builder $query) {
-                            $query->where('status', ServiceRequestAssignmentStatus::Active);
-                        })
+                        fn (Builder $query, array $data) => $query
+                            ->when(
+                                empty($data['value']),
+                                fn (Builder $baseQuery) => $baseQuery
+                            )
+                            ->when(
+                                $data['value'] === 'unassigned',
+                                fn (Builder $baseQuery) => $baseQuery->whereDoesntHave('assignedTo', function (Builder $query) {
+                                    $query->where('status', ServiceRequestAssignmentStatus::Active);
+                                })
+                            )
+                            ->when(
+                                ! empty($data['value']) && $data['value'] !== 'unassigned',
+                                fn (Builder $baseQuery) => $baseQuery->whereHas(
+                                    'assignedTo.user',
+                                    fn (Builder $userQuery) => $userQuery
+                                        ->where('id', $data['value'])
+                                )->whereHas(
+                                    'assignedTo',
+                                    fn (Builder $assignedToQuery) => $assignedToQuery
+                                        ->where('status', ServiceRequestAssignmentStatus::Active)
+                                )
```

Comment:
I could be wrong. But I don't think this check should be separate from the first `whereHas` here.

Does this result in a query that filters by users with that ID where the `ServiceRequestAssignmentStatus` is active?

We should probably adjust the test/s to ensure all of this is covered. So have Users where the assignment is active and ones where it is not, and then users where there is no assignment records at all.